main/TPS546.c

```diff
@@ -0,0 +1,748 @@
+#include "driver/i2c.h"
```

Comment:
you are doing the SMBus (PMBus) specific in the TPS546 driver. What would be nice is to make pmbus.c/.h in main, using i2c_master and exposing the API you need for the TPS546.